### PR TITLE
added book trip banner

### PR DESCRIPTION
**Pull request by sarahtsige**
- Related Issue (include '#'): #149 

- Description of changes made:  Added book trip banner to the existing banner component.

- Is the feature complete/bug resolved/etc..:  Feature is complete.  

- Any known bugs/strange behavior: No bugs.

- Is there specific feedback you would like on these changes:

- Screenshot(s):

Screen Shot 2020-04-06 at 4 10 22 PM

### CODE REVIEW

File: client/src/components/Banner/Banner.css [hunk 1]
```diff
@@ -18,3 +18,21 @@
   line-height: 72px;
   margin-bottom: 5vh;
 }
+.Banner-booktrip{
```
Comment (ZakkMan) [inline client/src/components/Banner/Banner.css:21]: `booktrip` is very contextual information. Is there a way to handle this without creating a whole new prop just for this one use case?

Comment (ZakkMan) [reply client/src/components/Banner/Banner.css:21]: Most of this CSS is probably unnecessary, right? It looks like banner can already handle having a background image. The only difference here is that the text has to be in the bottom right hand corner

File: client/src/components/Banner/Banner.css [hunk 2]
```diff
@@ -18,3 +18,21 @@
...
+  background-position: center;
+  background-size: cover;
+  width: 100vw;
+  color: #fafafa;
+  font-size: 100px;
+  font-family: var(--default-font);
+  position: relative;
+ 
+}
+.Banner-booktrip h3 {
```
Comment (ZakkMan) [inline client/src/components/Banner/Banner.css:32]: What if someone uses an `h2` or `h4`?

File: client/src/components/Banner/Banner.stories.js [hunk 3]
```diff
@@ -193,4 +194,14 @@ storiesOf("Layout/Banner", module)
...
+  .add("Book Trip Banner with Text", () => (
+    <Banner 
+    bannerType= "booktrip"
+    background={Banner__booktrip}>
+      <h3>Hawaii 2020</h3>
+    </Banner>
   ));
+
+
+  // style={{ textAlign: "right" }}
```
Comment (ZakkMan) [inline client/src/components/Banner/Banner.stories.js:207]: Can you delete this?

File: client/src/components/Banner/Banner.css (hunk 2 above)
Comment (sarahtsige) [reply client/src/components/Banner/Banner.css]: @ZakkMan  updated this to .Banner--large > *.   This should target the text no matter which tag is used.

File: client/src/components/Banner/Banner.stories.js (hunk 3 above)
Comment (sarahtsige) [reply client/src/components/Banner/Banner.stories.js]: @ZakkMan   Commented code has been deleted.

File: client/src/components/Banner/Banner.css (hunk 1 above)
Comment (sarahtsige) [reply client/src/components/Banner/Banner.css]: @ZakkMan changed from Banner-booktrip to class too Banner--large.

File: client/src/components/Banner/Banner.stories.js [hunk 4]
```diff
@@ -193,4 +194,18 @@ storiesOf("Layout/Banner", module)
         placeholder="Search Trip"
       />
     </Banner>
+  ))
+  .add("Book Trip Banner without Text", () => (
+    <Banner 
+    bannerType= "BannerLarge"
```
Comment (ZakkMan) [inline client/src/components/Banner/Banner.stories.js:200]: Could you make `bannerType`, `type`? That way it's more inline with the existing props. i.e. if you look at the `InputTextField` component, the prop is called `type`.

File: client/src/components/Banner/Banner.js [hunk 5]
```diff
@@ -2,7 +2,13 @@ import React from "react";
...
 
 const Banner = props => {
-  return <div className="Banner" style={{
+  let classList = '';
+  if (props.bannerType === 'BannerLarge' ) {
+    classList += ` Banner--large`
+  } else {
+    classList += ` Banner`
+  }
+  return <div className={classList} style={{
```
Comment (ZakkMan) [inline client/src/components/Banner/Banner.js:11]: `Banner--large` is a modifier, so it should be added in addition to `Banner`. I.e. `Banner Banner--large`

Comment (ZakkMan) [reply client/src/components/Banner/Banner.js:11]: This is where CSS becomes composable and lets you write dry CSS. I.e. all the properties you have repeated between these two classes can be removed no!

File: client/src/components/Banner/Banner.stories.js (hunk 4 above)
Comment (sarahtsige) [reply client/src/components/Banner/Banner.stories.js]: updated bannerType to type.

File: client/src/components/Banner/Banner.js (hunk 5 above)
Comment (sarahtsige) [reply client/src/components/Banner/Banner.js]: implemented recommendation above and removed repetitive CSS.

**Comment by saralaffin** [comment]
@clarknoah why was this closed?

**Comment by clarknoah** [comment]
@sarahtsige @saralaffin I get the following errors when try to compile the code: 
 Can't resolve 'components/BannerImageCard/BannerImageCard' in '/Users/clarknoah/Development/ga/sei/final_project/CoTripper_deployment_test/CoTrip/client/src/pages/Hawaii2020'

Please correct this so I can review!

**Comment by sarahtsige** [comment]
@clarknoah.   Let me know if you're still getting that error.  I pulled in the recent changes from develop so there should no longer be a merge conflict.  Thanks!

**Pull request merged**